pennylane/wires.py

```diff
@@ -0,0 +1,369 @@
+
+        Args:
+            list_of_wires (List[Wires]): list of Wires objects
+
+        Returns:
+            Wires: shared wires
+        """
+
+        for wires in list_of_wires:
+            if not isinstance(wires, Wires):
+                raise WireError(
+                    "Expected a `Wires` object; got {} of type {}.".format(wires, type(wires))
+                )
+
+        shared = []
+        # only need to iterate through the first object,
+        # since any wire not in this object will also not be shared
+        for wire in list_of_wires[0]:
+            if all(wire in wires_ for wires_ in list_of_wires):
+                shared.append(wire)
+
+        return Wires(shared)
+
+    @staticmethod
+    def all_wires(list_of_wires):
+        """Return the wires that appear in any of the Wires objects in the list.
+
+        This is similar to a set combine method, but keeps the order of wires as they appear in the list.
+
+        For example:
+
+        >>> wires1 = Wires([4, 0, 1])
+        >>> wires2 = Wires([3, 0, 4])
+        >>> wires3 = Wires([5, 3])
+        >>> list_of_wires = [wires1, wires2, wires3]
+        >>> Wires.all_wires(list_of_wires)
+        <Wires = [4, 0, 1, 3, 5]>
+
+        Args:
+            list_of_wires (List[Wires]): List of Wires objects
+
+        Returns:
+            Wires: combined wires
+        """
+
+        combined = []
+        for wires in list_of_wires:
+            if not isinstance(wires, Wires):
+                raise WireError(
+                    "Expected a `Wires` object; got {} of type {}".format(wires, type(wires))
+                )
+
+            combined.extend(wire for wire in wires.wire_tuple if wire not in combined)
+
+        return Wires(combined)
+
+    @staticmethod
+    def unique_wires(list_of_wires):
+        """Return the wires that are unique to any Wire object in the list.
+
+        For example:
+
+        >>> wires1 = Wires([4, 0, 1])
+        >>> wires2 = Wires([0, 2, 3])
+        >>> wires3 = Wires([5, 3])
+        >>> Wires.unique_wires([wires1, wires2, wires3])
+        <Wires = [4, 2, 5]>
+
+        Args:
+            list_of_wires (List[Wires]): list of Wires objects
+
+        Returns:
+            Wires: unique wires
+        """
+
+        for wires in list_of_wires:
+            if not isinstance(wires, Wires):
+                raise WireError(
+                    "Expected a `Wires` object; got {} of type {}.".format(wires, type(wires))
+                )
+
+        unique = []
+        for wires in list_of_wires:
+            for wire in wires:
+                # check that wire is only contained in one of the Wires objects
+                if sum([1 for wires_ in list_of_wires if wire in wires_]) == 1:
+                    unique.append(wire)
+
+        return Wires(unique)
+
+    @staticmethod
+    def merge(list_of_wires):
+        """Merge Wires objects in list to one. All Wires objects in the list must contain unique wires.
```

Comment:
So `Wires.merge(Wires([0, 4]), Wires([1, 4, 6]))` is not allowed, and instead `Wires.all_wires` should be used? I'm curious what the intended use case is for `merge` vs `all_wires`

Reply:
Was obsolete, forgot to delete!